src/main/java/org/spongepowered/api/entity/attribute/type/AttributeTypes.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.entity.attribute.type;
+
+import org.spongepowered.api.util.generator.dummy.DummyObjectProvider;
+
+public final class AttributeTypes {
+
+    public static final class Entity {
+
+        // SORTFIELDS:ON
+
+        public static final RangedAttributeType ARMOR = DummyObjectProvider.createFor(RangedAttributeType.class, "ARMOR");
+
+        public static final RangedAttributeType ARMOR_TOUGHNESS = DummyObjectProvider.createFor(RangedAttributeType.class, "ARMOR_TOUGHNESS");
+
+        public static final RangedAttributeType ATTACK_DAMAGE = DummyObjectProvider.createFor(RangedAttributeType.class, "ATTACK_DAMAGE");
+
+        public static final RangedAttributeType ATTACK_SPEED = DummyObjectProvider.createFor(RangedAttributeType.class, "ATTACK_SPEED");
+
+        public static final RangedAttributeType FOLLOW_RANGE = DummyObjectProvider.createFor(RangedAttributeType.class, "FOLLOW_RANGE");
+
+        public static final RangedAttributeType KNOCKBACK_RESISTANCE = DummyObjectProvider.createFor(RangedAttributeType.class, "KNOCKBACK_RESISTANCE");
+
+        public static final RangedAttributeType LUCK = DummyObjectProvider.createFor(RangedAttributeType.class, "LUCK");
+
+        public static final RangedAttributeType MAX_HEALTH = DummyObjectProvider.createFor(RangedAttributeType.class, "MAX_HEALTH");
+
+        public static final RangedAttributeType MOVEMENT_SPEED = DummyObjectProvider.createFor(RangedAttributeType.class, "MOVEMENT_SPEED");
+
```

Comment:
Should we also support the attribute `generic.reachDistance` which is added by forge?

Reply:
Is this something we can implement in vanilla? Does it need to be synced with the client?

Reply:
It will work nicer that way yes, but you can also make it server side only by blocking interaction if the reach distance is below the client distance, or handle custom interaction if it's the other way around (could also just limit this to forge).